ant-networking/src/event/swarm.rs

```diff
@@ -264,6 +266,15 @@ impl SwarmDriver {
                 event_string = "ConnectionClosed";
                 debug!(%peer_id, ?connection_id, ?cause, num_established, "ConnectionClosed: {}", endpoint_str(&endpoint));
                 let _ = self.live_connected_peers.remove(&connection_id);
+
+                if self.connected_relay_clients.contains(&peer_id) && num_established == 0 {
```

Comment:
just check `if num_established == 0` shall be enough, i.e. can change to
```rust
if num_established == 0 {
    self.connected_relay_clients.remove(&peer_id);
    info!(
        "Relay client has been disconnected: {peer_id:?}. Relay client count: {}",
        self.connected_relay_clients.len()
    );
}
```

Reply:
Made it into
```
if num_established == 0 && self.connected_relay_clients.remove(&peer_id) {
                    info!(
                        "Relay client has been disconnected: {peer_id:?}. Relay client count: {}",
                        self.connected_relay_clients.len()
                    );
                }
```